api/src/main/java/org/openmrs/validator/ConceptValidator.java

```diff
@@ -103,6 +105,9 @@ public void validate(Object obj, Errors errors) throws APIException, DuplicateCo
 			return;
 		}
 
+		ValidationUtils.rejectIfEmpty(errors, "datatype", "Concept.datatype.empty");
```

Comment:
You also need to add translations for these codes in the messages.properties file